includes/payment-methods/class-upe-payment-gateway.php

```diff
@@ -1068,4 +1086,45 @@ public function log_payment_error_ajax() {
 			);
 		}
 	}
+
+	/**
+	 * Creates a cookie to store the id and client secret of payment intent needed to mount the UPE element in frontend.
+	 *
+	 * An intent is tied to the cart's current state. New or removed items will likely impact the cart value,
+	 * necessitating the creation of a new intent with the revised cart value.
+	 * So a new cookie is expected to be created whenever the cart is modified.
+	 *
+	 * @param string $intent_id     The payment intent id.
+	 * @param string $client_secret The payment intent client secret.
+	 */
+	private function create_upe_payment_intent_cookie( string $intent_id = '', string $client_secret = '' ) {
+		$cart_hash = 'undefined';
+
+		if ( isset( $_COOKIE['woocommerce_cart_hash'] ) ) {
+			$cart_hash = sanitize_text_field( wp_unslash( $_COOKIE['woocommerce_cart_hash'] ) );
+		}
+
+		$cookie_val = $cart_hash . '-' . $intent_id . '-' . $client_secret;
+
+		wc_setcookie( self::COOKIE_UPE_PAYMENT_INTENT, $cookie_val, time() + MINUTE_IN_SECONDS * 30 );
+	}
+
+	/**
+	 * Creates a cookie to store the id and client secret of setup intent needed to mount the UPE element in frontend.
+	 *
+	 * @param string $intent_id     The setup intent id.
+	 * @param string $client_secret The setup intent client secret.
+	 */
+	private function create_upe_setup_intent_cookie( string $intent_id = '', string $client_secret = '' ) {
+		$cookie_val = $intent_id . '-' . $client_secret;
+		wc_setcookie( self::COOKIE_UPE_SETUP_INTENT, $cookie_val, time() + MINUTE_IN_SECONDS * 30 );
+	}
```

Comment:
My uneducated assumption was that there was a way to trigger the request to create a setup intent from the Checkout page. Based on your comments, it looks like this only happens on My Accounts pages or when checking out a subscription product with a free trial. There's certainly a lot to learn about the various possibilities and flows 😅 

At any rate, I wanted to keep different cookies for payment and setup intents because one depends on the cart's current state while the other does not. Also, it's a bit cleaner. Hence, two functions for two cookies.

Reply:
If you found my explanation reasonable, are we good to mark this conversation resolved?

Reply:
Cool! I don't really have a preference here, as long as it works. 😄 Keeping different cookies does sound reasonable enough, for the reasons you've shared. Resolving contentedly.